virtual_rainforest/models/abiotic_simple/simple_regression.py

```diff
@@ -0,0 +1,677 @@
+        dims=["layers", "cell_id"],
+        coords={
+            "layers": np.arange(0, len(interpolation_heights)),
+            "layer_roles": ("layers", layer_roles[0 : len(interpolation_heights)]),
+            "cell_id": mean_annual_temperature.cell_id,
+        },
+    )
+
+    return xr.concat(
+        [
+            DataArray(
+                np.full(
+                    (
+                        len(layer_roles) - len(interpolation_heights),
+                        len(mean_annual_temperature.coords["cell_id"]),
+                    ),
+                    np.nan,
+                ),
+                dims=["layers", "cell_id"],
+                coords={
+                    "layers": np.arange(len(interpolation_heights), len(layer_roles)),
+                    "layer_roles": (
+                        "layers",
+                        layer_roles[len(interpolation_heights) : len(layer_roles)],
+                    ),
+                    "cell_id": mean_annual_temperature.coords["cell_id"],
+                },
+            ),
+            layer_values_bound,
+        ],
+        dim="layers",
+    ).rename("soil_temperature")
+
+
+# TODO HYDROLOGY grid based in simple model?
+# Stream flow could be estimated as P-ET
+
+
+def calculate_soil_moisture(
+    layer_roles: List,
+    precipitation_surface: DataArray,
+    current_soil_moisture: DataArray,
+    soil_moisture_capacity: Union[DataArray, float] = MicroclimateParameters[
+        "soil_moisture_capacity"
+    ],
+) -> Tuple[DataArray, DataArray]:
+    """Calculate surface runoff and update soil mositure content.
+
+    Soil moisture and surface runoff are calculated with a simple bucket model: if
+    precipitation exceeds soil moisture capacity (see MicroclimateParameters), the
+    excess water is added to runoff and soil moisture is set to soil moisture capacity
+    value; if the soil is not saturated, precipitation is added to the current soil
+    moisture level and runoff is set to zero.
+
+    Args:
+        layer_roles: list of layer roles (from top to bottom: above, canopy, subcanopy,
+            surface, soil)
+        precipitation_surface: precipitation that reaches surface, [mm],
+        current_soil_moisture: current soil moisture at upper layer, [mm],
+        soil_moisture_capacity: soil moisture capacity (optional)
+
+    Returns:
+        current soil moisture, [mm], surface runoff, [mm]
+    """
+
+    available_capacity = soil_moisture_capacity - current_soil_moisture.mean(
+        dim="layers"
+    )
+
+    surface_runoff_cells = precipitation_surface.where(
+        precipitation_surface > available_capacity
+    )
+    surface_runoff = (
+        DataArray(surface_runoff_cells.data - available_capacity.data)
+        .fillna(0)
+        .rename("surface_runoff")
+        .rename({"dim_0": "cell_id"})
+        .assign_coords({"cell_id": current_soil_moisture.cell_id})
+    )
```

Comment:
Not fully sure, but I think this can be simply done as:
```suggestion
    surface_runoff = (
        (surface_runoff_cells.data - available_capacity.data)
        .fillna(0)
        .rename("surface_runoff")
    )
```